src/lib/API/Repository/Value/Filter/Filter.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Platform\API\Repository\Value\Filter;
+
+use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+use eZ\Publish\Core\Base\Exceptions\BadStateException;
+
+/**
+ * Content & Location filtering input Filter.
+ */
+final class Filter
+{
+    /** @var \eZ\Publish\API\Repository\Values\Content\Query\Criterion */
+    private $criterion;
+
+    /** @var \eZ\Publish\API\Repository\Values\Content\Query\SortClause */
+    private $sortClauses = [];
+
+    /**
+     * Build Filter.
+     *
+     * It's recommended to skip arguments and use `with...` and `andWith...` methods to build Filter.
+     *
+     * LocationService::createFindFilter() or ContentService::createFindFilter() can be used to
+     * instantiate Filter for better DX when creating chain of calls.
+     *
+     * @see \eZ\Publish\API\Repository\ContentService::createFindFilter()
+     * @see \eZ\Publish\API\Repository\LocationService::createFindFilter()
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException for invalid Sort Clause
+     */
+    public function __construct(?Criterion $criterion = null, array $sortClauses = [])
+    {
+        $this->criterion = $criterion;
+        foreach ($sortClauses as $idx => $sortClause) {
+            if (!$sortClause instanceof SortClause) {
+                throw new BadStateException(
+                    '$sortClauses',
+                    sprintf(
+                        'Expected an instance of "%s", got "%s" at position %d',
+                        SortClause::class,
+                        is_object($sortClause) ? get_class($sortClause) : gettype($sortClause),
+                        $idx
+                    )
+                );
+            }
+
+            $this->sortClauses[] = $sortClause;
+        }
+    }
+
+    /**
+     * Reset Filter so it can be built from scratch.
+     */
+    public function reset(): Filter
+    {
+        $this->criterion = null;
+        $this->sortClauses = [];
+
+        return $this;
+    }
+
+    /**
+     * Set filtering Criterion.
+     *
+     * If multiple Criteria are required, either use `andWithCriterion`/`orWithCriterion` or wrap
+     * them with Logical operator Criterion.
+     *
+     * To re-build Criterion from scratch `reset` it first.
+     *
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException if Criterion is already set
+     *
+     * @see reset
+     * @see andWithCriterion
+     * @see orWithCriterion
+     * @see \eZ\Publish\API\Repository\Values\URL\Query\Criterion\LogicalOr
+     * @see \eZ\Publish\API\Repository\Values\URL\Query\Criterion\LogicalAnd
```

Comment:
Did you mean `\eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalOr` and `\eZ\Publish\API\Repository\Values\Content\Query\Criterion\LogicalAnd` ?